circle.yml

```diff
@@ -15,3 +15,8 @@ test:
         files:
           - client/**/Makefile
           - server/**/Makefile
+    - case $CIRCLE_NODE_INDEX in 0) NODE_ENV=test make client/config/index.js && MOCHA_FILE=./test-results.xml npm run test-client -- --reporter=mocha-junit-reporter ;; 1) NODE_ENV=test make client/config/index.js && MOCHA_FILE=./test-results.xml npm run test-server -- --reporter=mocha-junit-reporter ;; esac:
+        parallel: true
+  post:
+    - mkdir -p $CIRCLE_TEST_REPORTS/junit/ && find . -type f -name "test-results.xml" -exec cp {} $CIRCLE_TEST_REPORTS/junit/ \;:
```

Comment:
CircleCi uploads them somewhere:
> PUT /27501-Automattic-wp-calypso-0/tmp/circle-junit.upR9vbT/junit/test-results.xml?AWSAccessKeyId=AKIAIQ65EYQDTMSJK2DQ&Expires=1458042646&Signature=moqTNYFwJMXHx0AV%2FlMrUybKCr8%3D HTTP/1.1

It works exactly the same with identical names, as well as when I added $CIRCLE_NODE_INDEX to the file name.